### PR TITLE
Habana DLC 1.4.0 Release

### PR DESCRIPTION
**Pull request by bukeao**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: habana/pytorch/buildspec-1-10-1.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
```
Comment (shantanutrip) [inline habana/pytorch/buildspec-1-10-1.yml:1]: We have made some changes in the way we store the older buildspec files. Can you please refer to the new naming convention?

File: habana/pytorch/training/docker/1.10/py3/synapseai1.4.0/Dockerfile.hpu [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance*
+
+RUN curl -o /license.txt https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.10/license.txt
```
Comment (shantanutrip) [inline habana/pytorch/training/docker/1.10/py3/synapseai1.4.0/Dockerfile.hpu:84]: Can we make this compatible with future releases. For eg, make an ARG `PT_VERSION=1.10` and use it in the URI - just to ensure that we do not end up releasing older license in the upcoming releases?

Comment (shantanutrip) [inline habana/pytorch/training/docker/1.10/py3/synapseai1.4.0/Dockerfile.hpu:84]: Can we make this compatible with future releases. For eg, make an ARG `PT_VERSION=1.10` and use it in the URI - just to ensure that we do not end up releasing older license in the upcoming releases?

File: habana/tensorflow/buildspec-2-8.yml [hunk 3]
```diff
@@ -1,8 +1,8 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
 region: &REGION <set-$REGION-in-environment>
 framework: &FRAMEWORK tensorflow
-version: &VERSION 2.5.0
```
Comment (shantanutrip) [inline habana/tensorflow/buildspec-2-8.yml:4]: Can we follow the new naming convention for the older buildspec files?

Comment (shantanutrip) [inline habana/tensorflow/buildspec-2-8.yml:4]: Can we follow the new naming convention for the older buildspec files?

File: test/dlc_tests/sanity/test_pre_release.py [hunk 4]
```diff
@@ -233,7 +233,7 @@ def test_framework_version_cpu(image):
         else:
             if "neuron" in image:
                 assert tag_framework_version in output.stdout.strip()
-            if all(_string in image for _string in ["pytorch", "habana", "synapseai1.3.0"]):
+            if all(_string in image for _string in ["pytorch", "habana", "synapseai1.4.0"]):
```
Comment (shantanutrip) [inline test/dlc_tests/sanity/test_pre_release.py:236]: Will it make it backward incompatible for SynAI 1.3? If yes, lets modify the condition

File: habana/pytorch/buildspec-1-10-1.yml (hunk 1 above)
Comment (bukeao) [reply habana/pytorch/buildspec-1-10-1.yml:1]: I'll just delete the build-1-10.yml and rename the buildspec-1-10-1.yml to buildspec-1-10.yml. I don't want to change the naming convention.

File: habana/pytorch/training/docker/1.10/py3/synapseai1.4.0/Dockerfile.hpu (hunk 2 above)
Comment (bukeao) [reply habana/pytorch/training/docker/1.10/py3/synapseai1.4.0/Dockerfile.hpu:84]: fixed.

Comment (bukeao) [reply habana/pytorch/training/docker/1.10/py3/synapseai1.4.0/Dockerfile.hpu:84]: duplicated

File: habana/tensorflow/buildspec-2-8.yml (hunk 3 above)
Comment (bukeao) [reply habana/tensorflow/buildspec-2-8.yml:4]: I think it's typo in this PR, has been fixed in 1.4.1 PR

Comment (bukeao) [reply habana/tensorflow/buildspec-2-8.yml:4]: I think it's typo in this PR, has been fixed in 1.4.1 PR

File: test/dlc_tests/sanity/test_pre_release.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (bukeao) [reply test/dlc_tests/sanity/test_pre_release.py:236]: I don't think we have to make it compatible for now, basically each version (1.3/1.4/1.4.1) has their own test code

Comment (bukeao) [reply test/dlc_tests/sanity/test_pre_release.py:236]: duplicated

**Comment by tejaschumbalkar** [comment]
Feel free to reopen if its been actively worked upon

**Pull request closed without merging**